src/web/utils/errorHandler.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { ErrorInfo, ReactNode } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+
+import { ApiError } from './apiError';
+
+import './errorHandler.scss';
+
+interface ErrorState {
+  errorId: string;
+  errorHash?: string;
+  errorCount: number;
+  hasError?: boolean;
+  message?: string;
+  stack?: string;
+  componentStack?: string;
+}
+
+export interface RenderedErrorProps {
+  errorId?: string;
+  errorHash?: string;
+  errorCount: number;
+  message?: string;
+  stack?: string;
+  componentStack?: string;
+}
+
+export function analyticsIdentifier(errorId?: string, errorHash?: string) {
+  if (!errorHash && !errorId) return 'No error identifier';
+  return errorHash
+    ? `error hash: ${errorHash}` // From backend
+    : `error ID: ${errorId}`; // Generated by frontend to make it possible to triage user reported errors.
+}
+
+export function ErrorView({
+  message,
+  errorId,
+  errorHash,
+}: {
+  message: string;
+  errorId?: string;
+  errorHash?: string;
+}) {
+  return (
+    <div className='error-content'>
+      <img alt='Error icon' src='uid2-logo.png' />
+      <div>Error</div>
+      <div>{message}</div>
+      <div>({analyticsIdentifier(errorId, errorHash)})</div>
+    </div>
+  );
+}
+export interface ErrorBoundaryPropType {
+  children: ReactNode;
+  [key: string]: unknown;
+}
+
+/** Create an Error Boundary around a given component.
+ * @param ErrorComponent - Custom error component used to display the error.
+ */
+export function errorHandler<P extends ErrorBoundaryPropType>(
+  ErrorComponent: React.ComponentType<RenderedErrorProps>
+) {
+  class ErrorHandlerWrapperComponent extends React.Component<P, ErrorState> {
+    ErrorComponent: React.ComponentType<RenderedErrorProps>; // eslint-disable-line react/no-unused-class-component-methods
+    constructor(props: P) {
+      super(props);
+      this.state = {
+        errorId: uuidv4() as string, // eslint-disable-line @typescript-eslint/no-unsafe-call
```

Comment:
If I remove the eslint-disable I don't get any error (same for the similar line further down) - do you need to disable the lint rule?

Reply:
Remove ✅ 



I get this if I remove [the one below](https://github.com/IABTechLab/uid2-self-serve-portal/pull/30/files#diff-dc7164d51566178ee537ff6c919a07c2c613e4a220a7ae81fa73a4563ea8558dR73) though.
```Unused method or property "ErrorComponent" of class "ErrorHandlerWrapperComponent"eslint[react/no-unused-class-component-methods](https://github.com/jsx-eslint/eslint-plugin-react/tree/master/docs/rules/no-unused-class-component-methods.md)```

I think its because linter can't figure out the dynamic calls that are made to the HOC.